torch/csrc/jit/codegen/cuda/kernel_cache.cpp

```diff
@@ -393,14 +393,20 @@ std::vector<at::Tensor> FusionKernelRuntime::runKernelWithInput(
     options.device = c10::Device(DeviceType::CUDA, device_index);
     FusionGuard fg(fusion_to_run.get());
     scheduler_entry->schedule(fusion_to_run.get());
-    executors_[group_id].compileFusion(fusion_to_run.get(), options);
-  }
-
-  // Load launch params for reduction and normalization kernels
-  if (scheduler_entry->hasReductionParam()) {
-    launch_params = scheduler_entry->reductionParams().lparams;
+    // Load launch params for reduction and normalization kernels
+    if (scheduler_entry->hasReductionParam()) {
+      launch_params = scheduler_entry->reductionParams().lparams;
+    } else {
+      launch_params = scheduler_entry->pointwiseParams().lparams;
+    }
+    executors_[group_id].compileFusion(fusion_to_run.get(), options, inputs, launch_params);
```

Comment:
Not loving this interface now that requires inputs and launch params for compilation.

Reply:
But this is only optional, as in:

<!--StartFragment-->  void compileFusion(Fusion* fusion, CompileOptions options = CompileOptions(),      const at::ArrayRef&lt;IValue&gt;&amp; inputs = {},      const LaunchParams&amp; launch_constraints = LaunchParams());<!--EndFragment-->

I agree this is still pretty ugly though :(
I only need to get the block size, but doing that requires a lowered kernel, which happens inside compileFusion. I can try to break it into two functions but I don't know if that would be a cleaner API